app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+class User < ActiveRecord::Base
+  has_many :posts, class_name: 'post', foreign_key: 'authors_id'
+  has_many :comments, class_name: 'comment', foreign_key: 'comments_id'
+  has_many :likes, class_name: 'like', foreign_key: 'likes_id'
+
+  def recent_posts
+    Post.where(authors_id: id).limit(3).order(created_at: :asc).pluck(:title, :text)
+  end
+
+  def update_posts_counter
+    user = self
+    user.increment!(:posts_counter)
+    user.save
+  end
```

Comment:
- [ ] As stated in the project requirement **A method that updates the posts counter for a user should be in Post models not in the User model so Kindly move it to the Post model**  and also in the Post model Kindly implemented the below optional Suggestion .
- [ Optional ] Kindly I recommend you to use  [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_posts_counter` method every time a new post has been saved to database, then make the `update_posts_counter` method private just like the below sample **after_save** is invoked every time a record is saved to the database, whether it is a new record or an existing record that is updated.

```
 def recent_comments
    Comment.where(posts_id: id).limit(5).order(created_at: :asc).pluck(:text)
  end

after_save :update_posts_counter

private
  def update_posts_counter
     author.posts_counter = author.posts.count
    author.save
  end

```